mpas_analysis/shared/timekeeping/MpasRelativeDelta.py

```diff
@@ -0,0 +1,154 @@
+                                  seconds=self.seconds + other.seconds,
+                                  calendar=self.calendar)
+
+        year = other.year+self.years
+
+        month = other.month
+        if self.months != 0:
+            assert 1 <= abs(self.months) <= 12
+            month += self.months
+            if month > 12:
+                year += 1
+                month -= 12
+            elif month < 1:
+                year -= 1
+                month += 12
+
+        if self.calendar == 'gregorian':
+            daysInMonth = monthrange(year, month)[1]
+        elif self.calendar == 'gregorian_noleap':
+            # use year 0001, which is not a leapyear
+            daysInMonth = monthrange(1, month)[1]
+
+        day = min(daysInMonth, other.day)
+        repl = {"year": year, "month": month, "day": day}
+
+        days = self.days
+        if self.calendar == 'gregorian_noleap' and isleap(year):
+            if month == 2 and day+days >= 29:
+                # skip forward over the leap day
+                days += 1
+            elif month == 3 and day+days <= 0:
+                # skip backward over the leap day
+                days -= 1
+
+        return (other.replace(**repl) +
+                datetime.timedelta(days=days,
+                                   hours=self.hours,
+                                   minutes=self.minutes,
+                                   seconds=self.seconds))
+
+    def __radd__(self, other):
+        return self.__add__(other)
+
+    def __rsub__(self, other):
+        return self.__neg__().__add__(other)
+
+    def __sub__(self, other):
+        if not isinstance(other, MpasRelativeDelta):
+            return NotImplemented
+        return self.__add__(other.__neg__())
+
+    def __neg__(self):
+        return self.__class__(years=-self.years,
+                              months=-self.months,
+                              days=-self.days,
+                              hours=-self.hours,
+                              minutes=-self.minutes,
+                              seconds=-self.seconds,
+                              calendar=self.calendar)
+
+    def __mul__(self, other):
+        try:
+            f = float(other)
+        except TypeError:
+            return NotImplemented
+
+        return self.__class__(years=int(self.years * f),
+                              months=int(self.months * f),
+                              days=int(self.days * f),
+                              hours=int(self.hours * f),
+                              minutes=int(self.minutes * f),
+                              seconds=int(self.seconds * f),
+                              calendar=self.calendar)
+
+    __rmul__ = __mul__
+
+    def __div__(self, other):
+        try:
+            reciprocal = 1 / float(other)
+        except TypeError:
+            return NotImplemented
+
+        return self.__mul__(reciprocal)
+
+    __truediv__ = __div__
+
+    def __repr__(self):
+        l = []
```

Comment:
This is dangerous for multiple reasons-- I would greatly prefer that we never use single-character variables unless they are an index.  Furthermore, "l" looks a lot like a "1", especially in certain fonts.  Cna you please make this something like `rep` or `strrep`, e.g. something three characters or more instead of `l`?